### PR TITLE
docs: add azure devops migration guide

### PR DESCRIPTION
**Pull request by hugorut**
* adds migration guide for azure devops integration.

Note: in the vast majority of cases I have opted to reference the marketplace extension over the github repo. This is because this is the actual entry/installation point for the user and I believe this makes the most sense.

### CODE REVIEW

File: docs/guides/azure_devops_migration.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+---
+slug: azure_devops_migration
+title: Azure DevOps migration
+---
+
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+Follow this page to migrate from our old [pipeline bash scripts](https://github.com/infracost/infracost-azure-devops/blob/0c662db3982f53666e89e69a406c572f48dc5c33/README.md) to our new [Azure Devops Extension](https://marketplace.visualstudio.com/items?itemName=Infracost.infracost-tasks) approach.
+
+<img src={useBaseUrl("img/screenshots/actions-pull-request.png")} alt="Cost estimate comment for multiple projects" />
```
Comment (hugorut) [inline docs/guides/azure_devops_migration.md:10]: waiting for Vadim to resolve `InfracostComment` issues so we can update this with new media.

File: docs/guides/azure_devops_migration.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
+---
+slug: azure_devops_migration
+title: Azure DevOps migration
+---
+
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+Follow this page to migrate from our old [pipeline bash scripts](https://github.com/infracost/infracost-azure-devops/blob/0c662db3982f53666e89e69a406c572f48dc5c33/README.md) to our new [Azure Devops Extension](https://marketplace.visualstudio.com/items?itemName=Infracost.infracost-tasks) approach.
```
Comment (aliscott) [inline docs/guides/azure_devops_migration.md:8]:
```suggestion
Follow this page to migrate from our old [pipeline bash scripts](https://github.com/infracost/infracost-azure-devops/blob/0c662db3982f53666e89e69a406c572f48dc5c33/README.md) to our new [Azure DevOps Extension](https://marketplace.visualstudio.com/items?itemName=Infracost.infracost-tasks) approach.
```

Comment (alikhajeh1) [reply docs/guides/azure_devops_migration.md:8]: I'd also drop the word "approach"

File: docs/guides/azure_devops_migration.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+<img src={useBaseUrl("img/screenshots/actions-pull-request.png")} alt="Cost estimate comment for multiple projects" />
+
+## What's new?
+
+🚀 The new [Infracost Azure DevOps Extension](https://marketplace.visualstudio.com/items?itemName=Infracost.infracost-tasks) provide a set of Azure Pipeline tasks offering a composable way of using Infracost in your pipeline. These JavaScript tasks simplify integrating Infracost into your Azure Pipeline. In addition, we've added Azure specific output formats, a cost summary table, and different behaviors, so you can control when comments are to be posted.
```
Comment (alikhajeh1) [inline docs/guides/azure_devops_migration.md:14]:
```suggestion
🚀 The new [Infracost Azure DevOps Extension](https://marketplace.visualstudio.com/items?itemName=Infracost.infracost-tasks) provide a set of Azure Pipeline tasks offering a composable way of using Infracost in your pipeline. These JavaScript tasks simplify integrating Infracost into your Azure Pipeline. In addition, we've added Azure specific output formats, a cost summary table, and different behaviors, so you can control when comments are posted.
```

I prefer fewer words but feel free to ignore this suggestion :)

File: docs/guides/azure_devops_migration.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+We've complied an expansive list of [examples](https://github.com/infracost/infracost-azure-devops#examples) demonstrating how these task can be used in different pipelines. 
+
+Composable tasks provide three key benefits:
+
+1. No need for a large setup steps: Installing the Infracost toolchain is now abstracted behind our nice new tasks, so there's no need for large "setup" steps that deal with pre-configuring Infracost dependencies. 
+2. Safe version upgrades: The InfracostSetup task has a `version` field for the CLI, which supports [SemVer ranges](https://www.npmjs.com/package/semver#ranges). So instead of a [full version](https://github.com/infracost/infracost/releases) string, you can use `0.9.x`. This enables you to automatically get the latest backward compatible changes in the 0.9 release (e.g. new resources/features and bug fixes) without worrying about CI/CD pipelines breaking.
+3. Versioning for the CI integration: Each of the Infracost tasks have a version, `InfracostSetup@0` & `InfracostComment@0`, which also supports Semver. So you can use `@0` to get backward compatible updates for the extension (e.g. bug fixes).
```
Comment (alikhajeh1) [inline docs/guides/azure_devops_migration.md:29]: Could these not be v1 like we have for the GH Actions? or is `@0` common for Azure extensions?

File: docs/guides/azure_devops_migration.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+### CI-specific formats
+
+The infracost output command now has two new format options: azure-devops-repo and slack-message. We already have formats for GitHub & Gitlab which are used by our [GitHub actions](https://github.com/infracost/actions) and [Gitlab](https://gitlab.com/infracost/infracost-gitlab-ci/) integrations.
```
Comment (alikhajeh1) [inline docs/guides/azure_devops_migration.md:33]:
```suggestion
The infracost output command now has two new format options: `azure-devops-repo` and `slack-message`. We already have formats for GitHub & Gitlab which are used by our [GitHub Actions](https://github.com/infracost/actions) and [GitLab CI](https://gitlab.com/infracost/infracost-gitlab-ci/) integrations.
```

**Review by alikhajeh1 — APPROVED** [review]
Love it! Thank you 🕺

**Comment by alikhajeh1** [comment]
@hugorut just FYI that I changed the target branch to v0.9.17 for this PR. In this repo, anything that gets merged to master goes live :)

File: docs/guides/azure_devops_migration.md (hunk 4 above)
Comment (hugorut) [reply docs/guides/azure_devops_migration.md:29]: `@0` is common in `Azure`

**Pull request merged**